lib/galaxy/tool_util/deps/mulled/util.py

```diff
@@ -79,14 +82,32 @@ def mulled_tags_for(namespace, image, tag_prefix=None):
 
 def split_tag(tag):
     """Split mulled image name into conda version and conda build."""
-    version = tag.split('--', 1)[0]
-    build = tag.split('--', 1)[1]
-    return version, build
+    return tag.split('--', 1)
+
+
+def parse_tag(tag):
+    """Split mulled image name into version and build number."""
+    version = tag
+    build_string = "-1"
+    if '--' in tag:
+        version, build_string = tag.rsplit('--', 1)
+    elif '-' in tag:
+        # Should be mulled multi-container image tag
+        version, build_string = tag.rsplit('-', 1)
+    build_number = int(BUILD_NUMBER_REGEX.search(tag).group(0))
```

Comment:
It would be safer to search on the build_string here, isn't it?

Reply:
that wouldn't work for mulled suffixes

Reply:
actually it might work, but I'd have to change the fallback and the regex, else -1 becomes 1